source/core/parameters_lagrangian.cc

```diff
@@ -393,14 +415,16 @@ namespace Parameters
       prm.enter_subsection("insertion info");
       {
         const std::string insertion = prm.get("insertion method");
-        if (insertion == "volume")
-          insertion_method = InsertionMethod::volume;
-        else if (insertion == "list")
-          insertion_method = InsertionMethod::list;
+        if (insertion == "clear_and_add")
+          insertion_method = InsertionMethod::clear_and_add;
```

Comment:
I am unsure this needs to be a seperate method, this is more a component of every insertion method. Do insertion methods clear before they add or not. That's pretty much the main idea.